Automattic/simplenote-electron
# Macos notarize

This PR adds notarization to the Mac build of Simplenote. 
In order to be notarized, the binary must be built with the `hardened runtime` flag. The Simplenote app seems to be ok with it when the proper entitlements are provided. 

Support for `hardened runtime` has been added on `electron-builder 20.41.0`, but I hit what seems to be a known bug with Simplenote Electron where versions of `electron-builder` higher than `20.32` break the Windows build. I found [this](https://github.com/electron-userland/electron-builder/issues/3654) issue by @mirka, but the problem hasn't been fixed yet. 

I did some investigation, but the issue seems quite complex, so I set up a temp stop-gap solution by patching `electron-builder` on the CI system after `npm install`. The code I'm using in the patch is the same I sent upstream (https://github.com/electron-userland/electron-builder/commit/7d5f952), so the configuration won't break when we manage to update `electron-build`. 
I don't like this solution too much, so if you have any other suggestion, I'm happy to hear it and to change this PR. 

The notarization process is performed on both the `.app` and the `.dmg` file in order to allow the users to install and start the app even when offline. 

The notarization process runs on tags. 

### Test
1. This is a build on a commit: https://circleci.com/gh/Automattic/simplenote-electron/6692. Check in the logs that the notarization has been skipped.
2. This is a build on a tag: https://circleci.com/gh/Automattic/simplenote-electron/6696. Check that the notarization has been performed and is successful.  
3. Verify that there's no sensible data lacked in the logs. 
4. Download the Mac artifacts and verify that the app still works as expected. 


### Release
The PR targets the release branch because Catalina has been released and, while Apple states that notarization won't be required until January 2020, there are reports of additional warnings already popping out that may be confusing for the users. 

These changes do not require release notes.
